src/App.tsx

```diff
@@ -14,37 +15,110 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+enum SortType {
+  BY_ALPH = 'alph',
+  BY_LENGTH = 'length',
+  DEFAULT = '',
+}
+
+type FilterProps = {
+  sortField: SortType;
+  reverse: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reverse }: FilterProps,
+) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods = preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.BY_ALPH:
+          return good1.localeCompare(good2);
+
+        case SortType.BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
```

Comment:
[CODE KNOWLEDGE] - Don't interact with DOM directly, use React as much as possible. The function getPreparedGoods is manipulating the data correctly for display, but ensure that all DOM interactions are handled through React. In this case, it's being done right, but keep this in mind for functions that might manipulate the DOM.